File: src/api/Domain/Property.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace Domain
{
    public class Property
    {
        [Key]
        public string Id { get; set; }

        [Required]
        public string Name { get; set; }

        [Required]
        public string Slug { get; set; }

        public string Suburb { get; set; }
        public string SuburbSlug { get; set; }
        public string State { get; set; }
        public string MainImage { get; set; }
        public string Thumbnail { get; set; }
        public string ShortDescription { get; set; }

        [Required]
        public int PropertyTypeId { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public int Bedrooms { get; set; }

        [Required]
        public int Bathrooms { get; set; }

        [Required]
        public int ParkingSpaces { get; set; }

        [Required]
        public decimal LandSize { get; set; }

        [Required]
        public string PostCode { get; set; }

        public string StreetNumber { get; set; }
        public string Route { get; set; }
        public string Locality { get; set; }
        public string AdministrativeAreaLevel1 { get; set; }
        public string AdministrativeAreaLevel2 { get; set; }

        [Required]
        public string Country { get; set; }

        [Required]
        public string CreatedById { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public bool IsActive { get; set; } = false; 
        public bool IsForSale { get; set; } = false;
        public bool IsForRent { get; set; } = false;
        public bool IsForAuction { get; set; } = false;
        public decimal BuyPrice { get; set; }
        public decimal RentPrice { get; set; }
        public DateTimeOffset AvailableDate { get; set; }

        public virtual PropertyType PropertyType { get; set; }
        
        public virtual ICollection<PropertyFeature> PropertyFeatures { get; set; }
        
        public virtual ICollection<Listing> Listings { get; set; }
        
        public virtual ICollection<Image> Images { get; set; }
        
        public virtual ICollection<UserPermission> UserPermissions { get; set; }

        public virtual ICollection<UserProperty> UserProperties { get; set; }
        
        public virtual ICollection<Activity> Activities { get; set; }
    }
}

File: src/ui/mgt/mgt-react-ts-sem/src/app/models/property.ts

export interface IProperty {
    id?: string;
    name?: string;
    slug?: string;
    suburb?: string;
    suburbSlug?: string;
    streetNumber?: string;
    route?: string;
    locality?: string;
    postCode?: string;
    administrativeAreaLevel1?: string;
    administrativeAreaLevel2?: string;
    state?: string;
    country?: string;
    mainImage?: string;
    propertyTypeId?: number;
    description: string;
    bedrooms?: number;
    bathrooms?: number;
    parkingSpaces?: number;
    landSize?: number;
    isActive?: boolean;
    isForSale?: boolean;
    isForRent?: boolean;
    isForAuction?: boolean;
    buyPrice?: number;
    rentPrice?: number;
    propertyType?: string;
    availableDate?: Date;
    propertyFeatures?: string[];
    propertyOffers?: any[];
    listings?: any[];
    images?: any[];
    activities?: any[]
}

export interface IPropertyFormValues extends Partial<IProperty>{
    time?: Date
}

export class PropertyFormValues implements IPropertyFormValues {
    id?: string = undefined;
    name: string =  '';
    suburb: string = '';
    postCode: string = '';
    availableDate?: Date = undefined;
    description: string = '';
    propertyTypeId?: number = undefined;
    bedrooms?: number = undefined;
    bathrooms?: number = undefined;
    parkingSpaces?: number = undefined;
    landSize?: number = undefined;
    streetNumber: string = '';
    route: string = '';
    locality: string = '';
    country: string = '';

    /**
     *
     */
    constructor(init?: IPropertyFormValues) {
        if (init && init.availableDate) {
            
        }
        Object.assign(this, init);

    }
}